### DISCUSSION TITLE
Preventing text from overflowing the box.

### DISCUSSION THREAD

**Discussion by wilsonowilson** [vercel/satori#336] [Q&A]
I'm trying to render a large paragraph of text with satori.

But, AFAIK, there's no way to dynamically adjust the text size to make sure it doesn't overflow the image, at least not without the DOM.

![image](https://user-images.githubusercontent.com/63465656/206433533-907d063e-fa1d-41d2-a209-979364a10b8e.png)

Does anybody have any ideas to achieve what I'm going for?

**Comment by altano** [comment]
I created a library, [@altano/satori-fit-text](https://www.npmjs.com/package/@altano/satori-fit-text), that solves this problem. It turns rendering into a two step process:

1. Use my library (which uses satori under the covers) to find the ideal font size for your container.
2. Take the font-size from step 1 and render out using satori just as you are already doing.

I have a write-up with examples here: https://alan.norbauer.com/articles/satori-fit-text

**Reply by altano** [reply]
@shuding this is such a common use case for satori, because it's a common use case for opengraph images. Would you be open to baking the behavior of my library directly into satori, even if that required non-standard css? Or are you only willing to allow standardized css in satori?
